exampleEnv/src/main/java/com/google/sps/Event.java

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Event is the container class for when a specific group of people are meeting and are therefore
+ * busy. Events are considered read-only.
+ */
+public final class Event {
+  private final String title;
+  private final TimeRange when;
+  private final Set<String> attendees = new HashSet<>();
+
+  /**
+   * Creates a new event.
+   *
+   * @param title The human-readable name for the event. Must be non-null.
+   * @param when The time when the event takes place. Must be non-null.
+   * @param attendees The collection of people attending the event. Must be non-null.
+   */
+  public Event(String title, TimeRange when, Collection<String> attendees) {
+    if (title == null) {
+      throw new IllegalArgumentException("title cannot be null");
+    }
+
+    if (when == null) {
+      throw new IllegalArgumentException("when cannot be null");
+    }
+
+    if (attendees == null) {
+      throw new IllegalArgumentException("attendees cannot be null. Use empty array instead.");
+    }
+
+    this.title = title;
+    this.when = when;
+    this.attendees.addAll(attendees);
+  }
+
+  /**
+   * Returns the human-readable name for this event.
+   */
+  public String getTitle() {
+    return title;
+  }
+
+  /**
+   * Returns the {@code TimeRange} for when this event occurs.
+   */
+  public TimeRange getWhen() {
+    return when;
+  }
+
+  /**
+   * Returns a read-only set of required attendees for this event.
+   */
+  public Set<String> getAttendees() {
+    // Return the attendees as an unmodifiable set so that the caller can't change our
+    // internal data.
+    return Collections.unmodifiableSet(attendees);
```

Comment:
if your intention is to have this entire class be read only, why not just store the unmodifiable set instead of making a copy each time?

Reply:
This code is part of the example environment that I am using to test code compilation. I did not write this class, I got it from the week 5 starter project. Should i still modify this class If I did not write it?

Reply:
I couldn't tell which code was written by you and which wasn't. feel free to leave it if it was just copied over